src/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -60,27 +69,67 @@ class VisEditor extends Component {
     }
   }
 
-  render() {
-    const handleChange = (part) => {
-      const nextModel = { ...this.state.model, ...part };
-
-      this.props.vis.params = nextModel;
-      if (this.state.autoApply) {
-        this.props.vis.updateState();
-      }
+  isIndexPatternChanged = () => {
+    if (this.state.model.index_pattern !== this.props.vis.params.index_pattern) {
+      return true;
+    }
+    const visAnnotations = this.props.vis.params.annotations || [];
+    const stateAnnotations = this.state.model.annotations || [];
+    const currentAnnotations = new Set(visAnnotations.map(annotation => annotation.index_pattern));
+    const prevAnnotations = new Set(stateAnnotations.map(annotation => annotation.index_pattern));
+    const diff = [...currentAnnotations].filter(a => !prevAnnotations.has(a));
+    return diff.length !== 0;
+  }
 
-      this.setState({ model: nextModel, dirty: !this.state.autoApply });
-    };
+  async fetchIndexPatternFields(vis) {
+    if (vis.params.index_pattern === '') {
```

Comment:
this feels like it doesn't belong here .... either we should set the default index pattern on init ... or when extracting the index patterns ? i uggest to move this logic inside extractIndexPatterns

Reply:
Right, I will check where is the good place to configure the default index pattern.